src/lensed-atom.ts

```diff
@@ -0,0 +1,28 @@
+import { IAtom, Atom } from './atom';
+import { Modifier } from './modifier';
+import { Projection } from './projection';
+import { Observable } from 'rxjs';
+import { over, view, compose } from 'ramda';
+import { Lens } from '@types/ramda';
+
+export class LensedAtom<T, S> implements IAtom<T> {
+  constructor(private lens: Lens, private source: Atom<S>) {
+  }
```

Comment:
It's not empty actually. `(private lens: Lens, private source: Atom<S>)` part sets two private properties on class instance (named as arguments: `lens` and `source`). TypeScript magic ;)